rasa/core/events/__init__.py

```diff
@@ -307,9 +307,10 @@ class BotUttered(Event):
 
     type_name = "bot"
 
-    def __init__(self, text=None, data=None, timestamp=None):
+    def __init__(self, text=None, data={}, metadata={}, timestamp=None):
```

Comment:
```suggestion
    def __init__(self, text=None, data=None, metadata=None, timestamp=None):
```
default values should *never* be `[]` or `{}`. the dict / array will only be created *once* and then reused for avery future call. this means if you then do `data["key"] = "value"` every future call of init will not have `{}` as a default value but `{"key": "value"}`.

Reply:
Good to know, I was wondering about that!